tt_metal/hw/firmware/src/brisc.cc

```diff
@@ -66,6 +65,11 @@ uint32_t tt_l1_ptr *rta_l1_base __attribute__((used));
 uint32_t tt_l1_ptr *crta_l1_base __attribute__((used));
 uint32_t tt_l1_ptr *sem_l1_base[ProgrammableCoreType::COUNT] __attribute__((used));
 
+uint16_t  dram_bank_to_noc_xy[NUM_NOCS][NUM_DRAM_BANKS] __attribute__((section(".uninit"), used));
```

Comment:
I added `used` attribute here, since we had it before  in genfiles.cpp/generated_bank_to_noc_coord.h
( we had this flag for the firmware part of the generated file). I will check if it's really needed, if not I will remove it.

Reply:
Looks like `used` attribute is needed. The kernel uses the symbols from firmware and accesses these arrays, which the firmware compiler doesn't know about.

Reply:
oh, I see.  Could you add a comment about that -- at the top of that whole section, so future-me is not confused?